### ISSUE TITLE
UserWarning: Implicit dimension choice for softmax has been deprecated.

### ISSUE THREAD

**Issue by gireeshkbogu** [qubvel/segmentation_models.pytorch#169]
How can I resolve this error? I'm using softmax for multi-class segmentation. I have 11 classes. So it should be something like `softmax(dim=10)`? 

`/root/.local/lib/python3.6/site-packages/segmentation_models_pytorch/base/modules.py:89: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  return self.activation(x)`

**Comment by da2r-20** [comment]
@gireeshkbogu can you share what was the resolution?

**Comment by vdplasthijs** [comment]
Also wondering this. In more detail; when defining a `smp` model in __init__() as:

```
self.base = smp.Unet(encoder_name='resnet50', pretrained='imagenet', 
                                  in_channels=3, classes=7,
                                  activation='softmax') 
```

This will throw the following warning upon initialisation:

`~/anaconda3/envs/geo/lib/python3.7/site-packages/segmentation_models_pytorch/base/modules.py:116: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  return self.activation(x)`

As the softmax is defined by passing 'softmax' as an arg, I'm not sure where/how to include the dim as the warning suggests? Many thanks

**Comment by kirillmeisser** [comment]
Hi, I have found a solution here https://github.com/qubvel/segmentation_models.pytorch/issues/689#issuecomment-1919000909.
